### ISSUE TITLE
Error during schema validation

### ISSUE THREAD

**Issue by sruedel-dotsource** [justinrainbow/json-schema#376]
I have the following schema:
`{
    "$schema": "http://json-schema.org/draft-04/schema#",
    "type": "object",
    "oneOf": [
        {
            "properties": {
                "localeId": {
                    "type": "integer"
                },
                "answer": {
                    "type": "string"
                },
                "inputs": {
                    "type": "array",
                    "items": {
                        "type": "object",
                        "$ref": "#/definitions/inputs"
                    }
                }
            },
            "required": [
                "localeId",
                "answer"
            ]
        },
        {
            "properties": {
                "answerId": {
                    "type": "integer"
                },
                "inputs": {
                    "type": "array",
                    "items": {
                        "type": "object",
                        "$ref": "#/definitions/inputs"
                    }
                }
            },
            "required": ["answerId"]
        }
    ],
    "definitions": {
        "inputs": {
            "properties": {
                "id": {
                    "type": "integer"
                },
                "value": {
                    "type": "string"
                }
            },
            "required": [
                "id",
                "value"
            ]
        }
    }
}`

If I want to validate data, the _JsonSchema\SchemaStorage::resolveRef_ were called with parameter _"#/definitions/inputs"_. The problem is that JsonPointer returned an empty filename  in _JsonSchema\SchemaStorage::addSchema_ and the _uriRetriever_ try to get contents from a empty file which runs into an error.

**Comment by erayd** [comment]
I'm happy to deal with this one - already trapping the error in 39569bf. Will go hunting for the root cause and squash it this weekend.

**Comment by erayd** [comment]
For anyone else following along, here's the schema from the original comment, with nicer formatting:
```JSON
{
    "$schema": "http:\/\/json-schema.org\/draft-04\/schema#",
    "type": "object",
    "oneOf": [
        {
            "properties": {
                "localeId": {
                    "type": "integer"
                },
                "answer": {
                    "type": "string"
                },
                "inputs": {
                    "type": "array",
                    "items": {
                        "type": "object",
                        "$ref": "#\/definitions\/inputs"
                    }
                }
            },
            "required": [
                "localeId",
                "answer"
            ]
        },
        {
            "properties": {
                "answerId": {
                    "type": "integer"
                },
                "inputs": {
                    "type": "array",
                    "items": {
                        "type": "object",
                        "$ref": "#\/definitions\/inputs"
                    }
                }
            },
            "required": [
                "answerId"
            ]
        }
    ],
    "definitions": {
        "inputs": {
            "properties": {
                "id": {
                    "type": "integer"
                },
                "value": {
                    "type": "string"
                }
            },
            "required": [
                "id",
                "value"
            ]
        }
    }
}
```

**Comment by erayd** [comment]
Document that reproduces the bug:
```JSON
{
    "answerId": 5,
    "inputs": []
}
```

**Comment by erayd** [comment]
Fixed in #378.

**Comment by bighappyface** [comment]
#378 is merged, this will be official in 6.0.0
